OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -165,11 +174,64 @@ static bool IsMovingInMyDirection(Actor self, Actor other)
 			return true;
 		}
 
+		public int TileSetMovementHash(TileSet tileSet)
+		{
+			var a = TilesetTerrainInfo[tileSet];
+			int hc = a.Length;
+			for (int i = 0; i < a.Length; i++)
+			{
+				hc = unchecked(hc * 31 + a[i].Cost);
+			}
```

Comment:
Use ```var``` where possible, and drop the braces here. Some more descriptive variable names for ```a``` and ```hc``` would be nice too.